### ISSUE TITLE
Tokio async flush stuck with big files

### ISSUE THREAD

**Issue by davxy** [helix-editor/helix#4301]
### Summary

The async flush hangs it the file lenght is greater than ~1M.

In particular this async call never returns:

https://github.com/helix-editor/helix/blob/661f48d62f9d6920f6929b5ae5a5365401459c61/helix-view/src/document.rs#L318

Reproducible creating a "big enough file" and exiting the editor with `:wq`.

After some tests I was able to find out the exact amount of bytes required to get the flush stuck.
This value (on machine) is exactly : 1,030,385 bytes

Note: the bug doesn't trigger if a write and quit in two separate actions.

### Reproduction Steps

1. Create a file big enough (1040385 bytes on my machine)

```
tr -dc A-Za-z0-9 < /dev/random | head -c 1040385 > test.txt
```

2. Open it in Helix.
3. Hit `:wq` in normal mode to exit

### Platform

Linux

### Terminal Emulator

Alacritty

### Helix Version

helix 22.08.1 (66276ce6)

**Comment by Ordoviz** [comment]
1040385 is exactly the amount of bytes that causes #3975, which will be fixed by #2267.

**Comment by the-mikedavis** [comment]
Duplicate of https://github.com/helix-editor/helix/issues/3975
